cli/verifier/no_shared_fs_mount.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2025 Edgeless Systems GmbH
+// SPDX-License-Identifier: BUSL-1.1
+
+package verifier
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/edgelesssys/contrast/internal/kuberesource"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	applycorev1 "k8s.io/client-go/applyconfigurations/core/v1"
+)
+
+// NoSharedFSMount verifies that no filesystems are shared with the root.
+type NoSharedFSMount struct{}
+
+// Verify verifies that no filesystems are shared with the root.
+func (v *NoSharedFSMount) Verify(toVerify any) error {
+	var findings error
+
+	// get all volume mounts that are referenced in containers
+	var volumeMounts []applycorev1.VolumeMountApplyConfiguration
+	kuberesource.MapPodSpec(toVerify, func(spec *applycorev1.PodSpecApplyConfiguration) *applycorev1.PodSpecApplyConfiguration {
```

Comment:
We probably need a `runtimeClassName` check here so that we don't complain about non-CC pods.